lib/peridiod/release/server.ex

```diff
@@ -440,9 +441,18 @@ defmodule Peridiod.Release.Server do
       {true, false} ->
         Logger.debug("Install Release: ok")
         Release.kv_progress(state.kv_pid, release_metadata)
-        state = do_binaries_jobs(binaries_metadata, Installer.Supervisor, callback, state)
-        state = %{state | installing_release: {release_metadata, binaries_metadata, callback}}
-        {:ok, state}
+
+        case binaries_metadata do
```

Comment:
this ensure that even if binaries_metadata got filtered to empty because everything was installed, that we still `finish_release`